c2/c2-client-bundle/c2-client-service/src/main/java/org/apache/nifi/c2/client/service/C2HeartbeatFactory.java

```diff
@@ -169,14 +171,20 @@ private NetworkInfo generateNetworkInfo() {
                     logger.debug("Instance has multiple interfaces.  Generated information may be non-deterministic.");
                 }
 
+                boolean networkInfoUnset = true;
                 for (NetworkInterface networkInterface : operationIfaces) {
                     Enumeration<InetAddress> inetAddresses = networkInterface.getInetAddresses();
-                    if (inetAddresses.hasMoreElements()) {
+                    while (inetAddresses.hasMoreElements()) {
                         InetAddress inetAddress = inetAddresses.nextElement();
-                        networkInfo.setDeviceId(networkInterface.getName());
-                        networkInfo.setHostname(inetAddress.getHostName());
-                        networkInfo.setIpAddress(inetAddress.getHostAddress());
-                        break;
+                        // IPv4 address is preferred over IPv6 as it provides more readable information for the user
+                        if (inetAddress instanceof Inet4Address) {
+                            updateNetworkInfo(networkInfo, networkInterface, inetAddress);
+                            break;
+                        }
+                        if (networkInfoUnset) {
+                            updateNetworkInfo(networkInfo, networkInterface, inetAddress);
+                            networkInfoUnset = false;
+                        }
```

Comment:
I think we should break the outer loop if any of these elements are found. Now the last networkInterface data is used instead of the first

Reply:
Thanks for the comment, good point. Made a change.